### ISSUE TITLE
In the External link admin page, distinguish between created github repo and utilised github repo E597

### ISSUE THREAD

**Issue by rija** [gigascience/gigadb-website#1142]
## User story

>As a curator  
>I want to add or edit Github links
>And specify whether those Github links are reference to data not in GigaDB or link representing data in GigaDB
>So that readers can clearly see which of those Github links are references

## Acceptance criteria

>Given I want associate a dataset with  a created GitHub repo
>When I navigate to the external link admin page
>Then I can add the GitHub link
>And I can select the "Created GitHub repo" type
>And I can save the external link

 >Given I want associate a dataset with  a referenced GitHub repo
>When I navigate to the external link admin page
>Then I can add the GitHub link
>And I can select the "Referenced GitHub repo" type
>And I can save the external link

## Additional Info

Part of Epic #597

It is related to #1137 and #1141

## Product Backlog Item Ready Checklist

* [ ] Business value is clearly articulated
* [ ] Item is understood enough by the IT team so it can make an informed decision as to whether it can complete this item
* [ ] Dependencies are identified and no external dependencies would block this item from being completed
* [ ] At the time of the scheduled sprint, the IT team has the appropriate composition to complete this item
* [ ] This item is estimated and small enough to comfortably be completed in one sprint
* [ ] Acceptance criteria are clear and testable
* [ ] Performance criteria, if any, are defined and testable
* [ ] The Scrum team understands how to demonstrate this item at the sprint review

## Product Backlog Item Done Checklist

* [ ] Item(s) in increment pass all Acceptance Criteria
* [ ] Code is refactored to best practices and coding standards
* [ ] Documentation is updated as needed
* [ ] Data security has not been compromised (with particular reference to the personal information we hold in GigaDB)
* [ ] No deviation from the team technology stack and software architecture has been introduced
* [ ] The product is in a releasable state (i.e. the increment has not broken anything)

**Comment by rija** [comment]
#1239, #331 and this one can be done at the same time (one migration script to bulk add those new external link. types)

**Comment by alli83** [comment]
@only1chunts @rija for the metadata to be sent to Datacite API when minting the DOI, how should we approach these new external link types?

**Comment by only1chunts** [comment]
@alli83 good point, thanks. For the first of them it already covered with the current actions (i.e. "created" github repo's are already included as "HasPart" e.g.
```
<relatedidentifier relatedidentifiertype="URL" relationtype="HasPart" resourcetypegeneral="Software">
      https://github.com/used-id/repo-name
```

For the new feature of adding referenced github links they should be added as "References":
```
<relatedidentifier relatedidentifiertype="URL" relationtype="References" resourcetypegeneral="Software">
      https://github.com/used-id/repo-name
```

**Comment by alli83** [comment]
@only1chunts I'm confused about the github links. In the dropdown, we should have "github links" (already exists) +  "Created GitHub repo" + "Referenced GitHub repo"?

**Comment by only1chunts** [comment]
no, just the two options "Created GitHub repo"(rename of already existing option) + "Referenced GitHub repo",

**Comment by alli83** [comment]
@only1chunts oh ok! thanks!

**Comment by pli888** [comment]
Closing this ticket because there is PR ##2263 for this work.

**Comment by alli83** [comment]
PR is still opened https://github.com/gigascience/gigadb-website/pull/2263
